packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx

```diff
@@ -47,31 +50,46 @@ type TreeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode"> &
  * @see https://itwinui.bentley.com/docs/tree
  * @alpha
  */
-export function TreeRenderer({
-  rootNodes,
+export function TreeRenderer({ rootNodes, selectNodes, selectionMode, rootErrorRenderer, ...treeProps }: TreeRendererProps) {
```

Comment:
Currently we allow `onNodeClick` and `onNodeKeyDown` to be passed into TreeRenderer which currently does nothing. How do you think we should handle this omit it or allow it to be passed and included in our `onNodeClick/onNodeKeyDown`? @saskliutas @grigasp 

We also allow various random properties like `onClick` which make no sense we should open an issue to clean up the API.

Reply:
> we should open an issue to clean up the API

we already have one - added a link to this comment to make sure it's covered: https://github.com/iTwin/presentation/issues/932#issuecomment-2826516743

Reply:
> Currently we allow `onNodeClick` and `onNodeKeyDown` to be passed into TreeRenderer which currently does nothing.

After a deeper look, I wouldn't say do nothing. As far as I can tell, they overwrite the callbacks that we create using `useSelectionHandler`. Which, I think, matches the pattern we have with other iTwinUI based components - if that's the case, I don't think we need to do anything about it. Adding @saskliutas for opinion.